### DISCUSSION TITLE
Implementation of auth.signUp() function in javascript

### DISCUSSION THREAD

**Discussion by PhucT1006** [supabase/supabase#21472] [Questions]
Hi, I'm new to creating websites, JS, and working with databases. I'm trying to write JS that will call the signUp() function provided by Supabase whenever the user clicks on the signup button. However when I try to call the signUp() auth function like below, nothing shows up on the auth user database on the Supabase website. I'm confused about whether this is the correct implementation of the auth signUp function. I've set up a create client file already. 

```
import supabase from "./configClient.js"

var sub = document.getElementById("button");
var em = document.getElementById("Email");
var pas = document.getElementById("Password");

sub.addEventListener( 'submit', ()=>{
  signUp();
}) 

async function signUp(){
let { data, error } = await supabase.auth.signUp({
  email:em.value,
  password: pas.value,
})
}


```

**Comment by GaryAustin1** [comment]
You should log out error from that call.

**Comment by silentworks** [comment]
My wild guess here is going to be that `em.value` are blank at the time of calling the function, its better to pass those in at the point of the form submission and not as reference when the code loads.

```js
import supabase from "./configClient.js"

var sub = document.getElementById("button");

sub.addEventListener('submit', async () => {
  var em = document.getElementById("Email");
  var pas = document.getElementById("Password");
  await signUp(em.value, pas.value);
}) 

async function signUp(email, password) {
  let { data, error } = await supabase.auth.signUp({
    email: email,
    password: password,
  })
}
```
Or even better you should be able to get the values from the form on submission
```js
import supabase from "./configClient.js"

var sub = document.getElementById("button");

sub.addEventListener('submit', async (event) => {
  event.preventDefault()
  const elements = new FormData(event.currentTarget)
  var em = elements.get("email_field_name");
  var pas = elements.get("password_field_name");
  await signUp(em, pas);
}) 

async function signUp(email, password) {
  let { data, error } = await supabase.auth.signUp({
    email: email,
    password: password,
  })

  if (error) {
    console.log({ error })
  }
  // do something to show the user their signUp was successful
}
```
Note the `email_field_name` would be the `name` attribute on the `email` `input` type field and same goes for password.

**Reply by vivilover** [reply]
this saved me! Thank you
